support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```diff
@@ -257,17 +258,26 @@ const StripeForm = (props: StripeFormPropTypes) => {
     }
   };
 
+  const handleCardSetupAndPay = () => handleCardSetup(setupIntentClientSecret)
+    .then(props.setStripePaymentMethod)
+    .then(() => {
+      setDisableButton(false);
+      props.submitForm();
+    })
+    .catch(() => {
+      setDisableButton(false);
+    });
+
   const requestSCAPaymentMethod = (event) => {
     event.preventDefault();
     props.validateForm();
     handleCardErrors();
     checkRecaptcha();
 
     if (stripe && props.allErrors.length === 0 && cardErrors.length === 0 && !recaptchaError) {
+      setDisableButton(true);
       if (setupIntentClientSecret) {
-        handleCardSetup(setupIntentClientSecret)
-          .then(props.setStripePaymentMethod)
-          .then(() => props.submitForm());
+        handleCardSetupAndPay();
       } else if (recaptchaCompleted) {
```

Comment:
From what I can see we only enable a disabled submit button (`setDisableButton(false)`) when the `handleCardSetupAndPay` call is made from within the `if` block of this `if...else if` statement. 

If the code were to go into the `else if` block here is there any other scenario that could follow where we'd need to enable a disabled submit button? Or at this point are we confident the only course of action is the form will submit successfully? For example, in this `else if` block we `setPaymentWaiting(true)`, however in the event the `handleStripeError` function is called we undo this (`setPaymentWaiting(false)`), would we also need to do the same for the submit button (`setDisableButton(false)`) in the event of a stripe error?

Reply:
If the `else if` block is entered, it means we're still waiting for `setupIntentClientSecret` to be available. When it becomes available the second `useEffect` hook will run, and if this `else if block _has_ been entered and set `paymentWaiting` to `true`, that hook will call `handleCardSetupAndPay`.

`handleStripeError` is only ever called in `handleCardSetup`, which is itself only called in `handleCardSetupAndPay`, which will always re-enable the button in the case of either success or error.

Reply:
OK thanks for clarifying this @i-hardy 👍